src/event_parser.py

```diff
@@ -26,17 +26,54 @@ def __init__(self, log_events, event_group_rules):
         self.event_groups_found = []
         self.event_groups_found_dict = {}
 
+    """ Initializes a search dictionary for tracking whether a group 
+        has attained all required events, a partial gorup will sit
+        in a queue with key of the next event it needs
+
+        ex: Group requires C, A, and T
+            Two partial groups exist; one with C and A, one with just C
+            search_dict: {
+                "C": []
+                "A": [[C]] # needs an A next
+                "T": [[C, A]] # needs a T next
+            } 
+            
+        Uses queues to ensure first occurences of events are grouped
+        together """
     def get_event_search_dict(self, event_group_rule):
         search_dict = {tag: [] for tag in event_group_rule.trigger_event_tags}
         return search_dict
 
+    """ Moves partial groups from stage to stage in the search dict,
+        or stores them as valid if they complete """
     def update_search_dict(self, group_tag, search_dict, event, context_event_buffer):
         ordered_tags = list(search_dict.keys())
         event_tag = event.tag
         tag_index = ordered_tags.index(event_tag)
         next_tag_index = tag_index + 1
 
-        if tag_index == 0:
+        if tag_index == 0 and next_tag_index == len(ordered_tags):
+            """ In the special case that a group only requires one event
```

Comment:
Docstrings are really only for modules, classes and functions. These should just be normal comments.